query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/QueryRequestEligibilityValidator.java

```diff
@@ -89,12 +90,14 @@ private boolean selectionAndGroupByOnDifferentColumn(
     Set<String> selections =
         selectionList.stream()
             .map(QueryRequestUtil::getLogicalColumnName)
+            .map(Optional::get)
```

Comment:
If you want it to error on unexpected
```suggestion
            .map(Optional::orElseThrow)
```
If you want it to drop and recover on unexpected
```suggestion
            .flatMap(Optional::stream)
```

Reply:
resolved